flight_planning/v1/flight_planning.yaml

```diff
@@ -0,0 +1,371 @@
+    attempting to plan, update, and close flight plans.
+    
+    The client may also (with a separate authorization scope) act as the test director to determine the status of the
+    USS's system under test, and request that the test area be cleared of any dangling flight plans.
+
+components:
+  securitySchemes:
+    Authority:
+      type: oauth2
+      flows:
+        clientCredentials:
+          tokenUrl: https://auth.example.com/oauth/token
+          scopes:
+            interuss.flight_planning.direct_automated_test: |-
+              Client may determine the test-readiness of the USS and instruct the USS to perform various actions to
+              prepare for tests involving flight planning (acting as test director).
+            interuss.flight_planning.plan: |-
+              Client may act as a user attempting to plan a flight, modify an existing flight plan, or close an existing
+              flight plan.
+      description: |-
+        Authorization from, or on behalf of, an authorization authority, augmenting standard strategic conflict detection or other similar authorization for the purpose of automated testing.
+
+  schemas:
+    StatusResponse:
+      type: object
+      required:
+        - status
+      properties:
+        status:
+          description: >-
+            The status of the USS automated testing interface.
+
+            - `Starting`: the USS is starting and the automated test driver should wait before sending requests.
+
+            - `Ready`: the USS is ready to receive test requests.
+          type: string
+          enum: [Starting, Ready]
+          example: Ready
+        system_version:
+          description: |-
+            Arbitrary string representing the version of the USS system to be tested.
+          type: string
+          example: v0.0.1-445ad3
+        api_name:
+          description: |-
+            Indication of the API implemented at this URL.  Must be "Flight Planning Automated Testing Interface".
+          type: string
+          example: Flight Planning Automated Testing Interface
+        api_version:
+          description: |-
+            Indication of the API version implemented at this URL.  Must be "v0.1.0" when implementing this version of the API.
+          type: string
+          example: v0.1.0
+
+    FlightPlan:
+      description: >-
+        Details of user's intent to create or modify a flight plan.
+      required:
+        - basic_information
+      type: object
+      properties:
+        basic_information:
+          $ref: '#/components/schemas/BasicFlightPlanInformation'
+        astm_f3548_21:
+          $ref: './scd.yaml#/components/schemas/ASTMF354821OpIntentInformation'
+        uspace_flight_authorisation:
+          $ref: './uspace.yaml#/components/schemas/FlightAuthorisationData'
+        au:
+          $ref: './au.yaml#/components/schemas/AUFlightDetails'
```

Comment:
I assume those are mutually exclusive. Would it make sense putting them under a `oneOf`?

Reply:
I agree it would be very unlikely to have a use case where both `uspace_flight_authorisation` and `au` were populated, but I don't know that we need to enforce anything.  This data will be populated by uss_qualifier based on test data provided by test designers, so presumably it should already be reasonable to use.

Many code generation tools do not deal well with polymorphism, so the best practice to maximize cross-platform compatibility is to never use oneOf to allow multiple data types for a single field.  I do not believe there is a way in OpenAPI 3.0 to specify that one of two different fields must be populated (though this is a super handy construct and used extensively in gRPC/protobuf).

Reply:
> Many code generation tools do not deal well with polymorphism, so the best practice to maximize cross-platform compatibility is to never use oneOf to allow multiple data types for a single field.

I did not consider this aspect, but indeed that make a lot of sense.